flycheck-rust.el

```diff
@@ -159,6 +163,20 @@ Flycheck according to the Cargo project layout."
                     (list (expand-file-name "target/debug" root)
                           (expand-file-name "target/debug/deps" root)))))))
 
+;;;###autoload
+(defun flycheck-rust-explain-error (error-code)
+  "Explain ERROR-CODE by invoking `rustc --explain'.
+
+ERROR-CODE defaults to the code of the error under point."
+  (interactive
+   (list (let ((errors-at-point (flycheck-overlay-errors-at (point))))
+           (when (> (length errors-at-point) 0)
```

Comment:
Please don't use `length` to check for non empty lists.  It's redundant and inefficient.  Instead exploit the fact that a non empty list is non-nil and just write `(and errors (flycheck-error-id errors))`.